odoo/addons/test_assetsbundle/tests/test_assetsbundle.py

```diff
@@ -632,3 +660,340 @@ def test_01_debug_mode_assets(self):
+            'name': 'test_jsfile4',
+            'bundle': 'test_assetsbundle.manifest4',
+            'directive': 'replace',
+            'glob': 'test_assetsbundle/static/src/js/test_jsfile1.js',
+            'target': 'test_assetsbundle/static/src/js/test_jsfile3.js',
+        })
+        view._render()
+        attach = self.env['ir.attachment'].search([('name', 'ilike', 'test_assetsbundle.manifest4.js')], order='create_date DESC', limit=1)
+        content = attach.raw.decode()
+        self.assertStringEqual(
+            content,
+            '''
+            /* /test_assetsbundle/static/src/js/test_jsfile1.js defined in bundle 'test_assetsbundle.manifest4' */
+            var a=1;
+            '''
+        )
+
+    def test_07_remove(self):
+        view = self.make_asset_view('test_assetsbundle.manifest5')
+        self.env['ir.asset'].create({
+            'name': 'test_jsfile4',
+            'bundle': 'test_assetsbundle.manifest5',
+            'directive': 'remove',
+            'glob': 'test_assetsbundle/static/src/js/test_jsfile2.js',
+        })
+        view._render()
+        attach = self.env['ir.attachment'].search([('name', 'ilike', 'test_assetsbundle.manifest5.js')], order='create_date DESC', limit=1)
+        content = attach.raw.decode()
+        self.assertStringEqual(
+            content,
+            '''
+            /* /test_assetsbundle/static/src/js/test_jsfile1.js defined in bundle 'test_assetsbundle.manifest5' */
+            var a=1;;
+
+            /* /test_assetsbundle/static/src/js/test_jsfile3.js defined in bundle 'test_assetsbundle.manifest5' */
+            var c=3;;
+
+            /* /test_assetsbundle/static/src/js/test_jsfile4.js defined in bundle 'test_assetsbundle.manifest5' */
+            var d=4;
+            '''
+        )
+
+    def test_08_remove_error(self):
+        view = self.make_asset_view('test_assetsbundle.manifest5')
+        self.env['ir.asset'].create({
+            'name': 'test_jsfile4',
+            'bundle': 'test_assetsbundle.manifest5',
+            'directive': 'remove',
+            'glob': 'test_assetsbundle/static/src/js/test_doesntexist.js',
+        })
+        # LPE Fixme: find an exception type
+        with self.assertRaises(Exception) as cm:
+            view._render()
+        self.assertTrue(
+            "File test_assetsbundle/static/src/js/test_doesntexist.js not found" in cm.exception.message
+        )
+
+    def test_09_remove_wholeglob(self):
+        view = self.make_asset_view('test_assetsbundle.manifest2')
+        self.env['ir.asset'].create({
+            'name': 'test_jsfile4',
+            'bundle': 'test_assetsbundle.manifest2',
+            'directive': 'remove',
+            'glob': 'test_assetsbundle/static/src/*/**',
+        })
+        view._render()
+        attach = self.env['ir.attachment'].search([('name', 'ilike', 'test_assetsbundle.manifest2.js')], order='create_date DESC', limit=1)
+        # LPE Fixme: is this really what we want ?
```

Comment:
@Arcasias  here too

Reply:
It doesn't seem to be working here... I would've expected the whole `src` directory to be wiped from the resulting assets. Is it perhaps because you used `*/**` instead of `**/*`? Not being an expert in globs I don't even know if there is a difference